util/dvsim/SimCfg.py

```diff
@@ -646,40 +645,51 @@ def create_failure_message(test, line, context):
         return results_str
 
     def gen_results_summary(self):
+        '''Generate the summary results table.
 
-        # sim summary result has 5 columns from each SimCfg.results_summary
-        header = ["Name", "Passing", "Total", "Pass Rate"]
-        if self.cov_report_deploy is not None:
-            header.append('Coverage')
-        table = []
-        colalign = ("center", ) * len(header)
-        for item in self.cfgs:
-            row = []
-            for title in item.results_summary:
-                row.append(item.results_summary[title])
-            if row:
-                table.append(row)
-        self.results_summary_md = "## " + self.results_title + " (Summary)\n"
-        self.results_summary_md += "### " + self.timestamp_long + "\n"
+        This method is specific to the primary cfg. It summarizes the results
+        from each individual cfg in a markdown table.
+
+        Prints the generated summary markdown text to stdout and returns it.
+        '''
+
+        text = [f"## {self.results_title} (Summary)"]
+        text += [f"### {self.timestamp_long}"]
```

Comment:
Nit: For single additions, I'd probably use `append`. It's probably slightly more efficient (because it doesn't construct a temporary singleton list), not that this really matters.

Reply:
I thought `extend` op was O(1) if the new list being appended has only 1 item. This feels a bit cleaner to me.

Reply:
Cool with me. Well, neither option is strictly O(1) (because Python lists are contiguous, so an append might imply a re-allocation) but you're right that there's no complexity difference between an append and an extension with a constant-length list.

Looking a bit more closely than the complexity class, the `+=` version of the code needs to construct a singleton list, pull out its only element and append it to `text` and then throw the list away again. I wouldn't think that the Python runtime is clever enough to fold that away to just an append.

Of course, this is Python code. Arguably, we're Doing It Wrong if we're worrying about performance :smile:. So I agree: go with whatever looks clearer!